fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -1066,6 +1067,23 @@ class BrowserRobot {
             return HomeScreenRobot.Transition()
         }
 
+        fun goToHomescreenWithComposeTopSites(composeTestRule: HomeActivityComposeTestRule, interact: ComposeTopSitesRobot.() -> Unit): ComposeTopSitesRobot.Transition {
+            clickPageObject(itemWithDescription("Home screen"))
+
+            mDevice.findObject(UiSelector().resourceId("$packageName:id/homeLayout"))
+                .waitForExists(waitingTime) ||
+                mDevice.findObject(
+                    UiSelector().text(
+                        getStringResource(R.string.onboarding_home_screen_jump_back_contextual_hint_2),
+                    ),
+                ).waitForExists(waitingTime)
+
+            composeTestRule.onNodeWithTag(TopSitesTestTag.topSites).assertExists()
```

Comment:
I would remove this check in case there's a scenario in which the top sites are disabled. (it will always try to verify the list exists even if the setting is disabled)